R/process_data.R

```diff
@@ -62,6 +68,59 @@ process_pacta_results <- function(data, start_year, end_year, time_horizon,
   return(data_processed)
 }
 
+#' Remove rows from PACTA results that belong to company-technology combinations
+#' for which there is no information on the exposure in the portfolio. We join
+#' company results on the portfolio exposure based on the last available year of
+#' the production forecast. Hence we filter for missings in that year.
+#'
+#' @inheritParams calculate_annual_profits
+#' @inheritParams report_company_drops
+#' @param data tibble containing filtered PACTA results
+#'
+#' @return A tibble of data without rows with no exposure info
+#' @noRd
+remove_companies_with_missing_exposures <- function(data,
+                                                    start_year,
+                                                    time_horizon,
+                                                    log_path) {
+  n_companies_pre <- length(unique(data$company_name))
+
+  companies_missing_exposure_value <- data %>%
+    dplyr::filter(.data$year == .env$start_year + .env$time_horizon) %>%
+    dplyr::filter(is.na(.data$plan_carsten)) #%>%
+
+  data_filtered <- data %>%
+    dplyr::anti_join(
+      companies_missing_exposure_value,
+      by = c("company_name", "technology")
```

Comment:
there are 4 companies with missing exposure value. you observed the drops for 2 of the companies only, do you know what happened to the other 2

Reply:
yes, they have 0 production plans in the last year, while they have positive ones in the beginning, both for high carbon techs. They were previously removed as that is considered a technology phase out. So in both cases they are correctly not part of the aggregation

Reply:
the other 2 companies that previously slipped through, had positive production targets at the end of the forecast, hence they were not considered to phase out the technologies and in that case we must have an exposure.